### PR TITLE
add data structures practice task (draft)

### PR DESCRIPTION
**Pull request by AMashoshyna**

### CODE REVIEW

**Comment by lempiy** [comment]
Maybe add binary tree with BFS ?
I think queue and stack is not enough.
And probably we don't need both queue and stack since difference is only in FIFO/LIFO.
I suggest to replace queue with BFS, since queue is used in BFS algorithm.
Also would be nice to have linked lists here.

So my suggestion is:
1. Stack. (add, get)
2. LinkedList (insert, show all)
3. Binary Tree (create, insert, find)

**Comment by AMashoshyna** [comment]
@lempiy  Good idea, thanks! I will update the task.

File: tasks/data-structures.md [hunk 1]
```diff
@@ -4,6 +4,36 @@
 
 # Data Structures
 
+
+# Practice
+
+Remember the http module? Time for practice! Write an application which implements stack and queue with access over http.
+A consumer of the app may add items to stack or queue and retrieve items. Stack and queue may contain only numbers and strings. There should be at least two different endpoints (one for stack, one for queue), everything else is up to you. Apply your knowledge of http to decide about appropriate http verbs to implement stack/queue functionality. Decide about the behavior of your app when stack or queue is empty.
```
Comment (intrueder) [inline tasks/data-structures.md]:
```suggestion
A consumer of the app may add items to stack or queue and retrieve items one by one. Stack and queue may contain only numbers and strings. There should be at least two different endpoints (one for stack, one for queue), everything else is up to you. Apply your knowledge of http to decide about appropriate http verbs to implement stack/queue functionality. Decide about the behavior of your app when stack or queue is empty.
```

**Comment by intrueder** [comment]
Hi. I have some suggestions.

When you are saying the stack should have `get` operation, and taking into account that this operation must remove an item from collecton, actual purpose of the operation will be treated as side effect. Also, web programmers may confuse `get` name  with http GET verb, which is idempotent operation, but for stack it is not. So, please, use `push`/`pop` terms (and, optionally, `peek`) when describing operations on stack https://en.wikipedia.org/wiki/Stack_(abstract_data_type)

As for linked lists, I'd suggest to add `remove` operation, as it is one of operations when linked list shows its best.

**Comment by lempiy** [comment]
@intrueder sounds reasonable.

File: tasks/data-structures.md [hunk 2]
```diff
@@ -4,9 +4,61 @@
 
 # Data Structures
 
+# Practice
+
+Remember the http module? Time for practice! Write an application which implements stack and linked list over http.
+A consumer of the app may add items to stack or linked list and retrieve items one by one. Stack and linked may contain only numbers and strings. There should be at least two different endpoints (one for stack, one for linked list), everything else is up to you. Apply your knowledge of http to decide about appropriate http verbs to implement stack/linked list functionality. Decide about the behavior of your app when stack or linked list is empty.
```
Comment (AMashoshyna) [inline tasks/data-structures.md]:
```suggestion
A consumer of the app may add items to stack or linked list and retrieve items one by one. Stack and linked list may contain only numbers and strings. There should be at least two different endpoints (one for stack, one for linked list), everything else is up to you. Apply your knowledge of http to decide about appropriate http verbs to implement stack/linked list functionality. Decide about the behavior of your app when stack or linked list is empty.
```

**Pull request merged**